common/src/include/ebbrt/MulticoreEbb.h

```diff
@@ -11,10 +11,42 @@
 
 #include <ebbrt/Cpu.h>
 #include <ebbrt/EbbAllocator.h>
+#include <ebbrt/EbbId.h>
 #include <ebbrt/EbbRef.h>
 #include <ebbrt/LocalIdMap.h>
 
 namespace ebbrt {
+
+template <class T> class MulticoreEbbRoot {
+ public:
+  struct RepMapRef {
+    LocalIdMap::ConstAccessor accessor;
+    const boost::container::flat_map<size_t, T*>* ref;
+  };
```

Comment:
I'd rather see just the accessor as a private member, and a method (or the dereference operator) return a reference to the map rather than store it.